src/libslic3r/PrintConfig.hpp

```diff
@@ -789,6 +789,8 @@ PRINT_CONFIG_CLASS_DEFINE(
     ((ConfigOptionPercent,            tree_support_top_rate))
     ((ConfigOptionFloat,              tree_support_branch_diameter_organic))
     ((ConfigOptionFloat,              tree_support_branch_angle_organic))
+    ((ConfigOptionBool,               enable_gap_fill_for_solid_infill))
```

Comment:
These two options seems duplicated with `filter_out_gap_fill` option.
Shall we combine these two into one enum option, something like: `gap_filter_target`?

Reply:
Isn’t the filter out a number though? What are you considering?

Reply:
Ah I got you. The two bools into a single enum. I’ll update it tomorrow.

Reply:
To make the logic consistent, we suggest we always enable gap fill for surface/infill
Then add the proposed new option which allow users filter out gap-fillings on specified area(all/internal)

Reply:
> To ensure consistency in the logic, I suggest always enabling gap fill for surfaces and infill. Then, we can add the proposed new option, which allows users to filter out gap fillings in specified areas (either all or internal only).

Understood. So an option that is called gap_filter_target, named in the UI: Don’t apply gap fill, with options: everywhere, top and bottom surfaces, infill. Correct?

Reply:
Also when everywhere is selected we can hide the filter small gap fill number

Reply:
> Also when everywhere is selected we can hide the filter small gap fill number

The current logic of `filter_out_gap_fill` is this: when it's set to 0, we don't filter any gaps. Conversely, when it's set to a specific number, for example, 6.5, all gap fills shorter than this value will be removed. 

Therefore, it's the other way around: when filter_out_gap_fill is set to 0, we hide gap_filter_target.